File: hr-cpp/hr-c6-basemanage/service/JobTitleInfo/JobTitleInfo.cpp
#include "stdafx.h"
#include "JobTitleInfo.h"
#include "../../dao/JobTitleInfo/JobTitleInfoDAO.h"
#include "../../uselib/excel/ExportExcel.h"
#include "../../uselib/fastdfs/UseFastDfs.h"

std::string JobTitleInfoService::listAllJobTitle(const JobTitleDTO::Wrapper& query)
{
	std::string url;
	JobTitleInfoDAO dao;
	auto countQuery = JobTitleQuery::createShared();
	countQuery->employee_id = query->employee_id;
	countQuery->employee_name = query->employee_name;
	countQuery->org_name = query->org_name;
	countQuery->jobtitle_name = query->jobtitle_name;
	countQuery->jobtitle_grades = query->jobtitle_grades;
	countQuery->b_highest_professional_title = query->b_highest_professional_title;
	uint64_t count = dao.count(countQuery);
	if (count <= 0)
	{
		return url;
	}

	list<JobTitleDO> resultDO = dao.selectAll(query);
	vector<vector<string>> data;
	// 将DO转换成二维数组
	for (auto const& sub : resultDO)
	{
		vector<string> temp;
		temp.push_back(sub.getEmployee_Id());
		temp.push_back(sub.getEmployee_Name());
		temp.push_back(sub.getOrg_Name());
		temp.push_back(sub.getJobtitle_Name());
		temp.push_back(sub.getGet_Time());
		temp.push_back(sub.getProfessional_Cate());
		temp.push_back(sub.getJobtitle_Grades());
		temp.push_back(sub.getTitle_Employment_Time());
		temp.push_back(sub.getIssuing_Authority());
		temp.push_back(sub.getJudging_Unit());
		temp.push_back(sub.getB_Highest_Professional_Title());
		data.push_back(temp);
	}

	// 生成数据表表头
	vector<string> head = dao.getHead(query);
	//head.erase(head.begin() + 12,head.end());
	data.insert(data.begin(), head);
	// 生成Excel
	ExportExcel excel;
	string filename = excel.exportExcel(data);
	UseFastDfs dfs("8.130.87.15");
	url = dfs.uploadWithNacos(filename);
	return url;
}

JobTitleDTO::Wrapper JobTitleInfoService::queryDataDetail(const JobTitleDTO::Wrapper& query)
{
	JobTitleInfoDAO dao;
	auto resD0 = dao.selectAll(query);
	auto resDTO = JobTitleDTO::createShared();
	resDTO->id = resD0.front().getId();
	resDTO->employee_id = resD0.front().getEmployee_Id();
	resDTO->employee_name = resD0.front().getEmployee_Name();
	resDTO->employee_state = resD0.front().getEmployee_State();
	resDTO->org_name = resD0.front().getOrg_Name();
	resDTO->get_time = resD0.front().getGet_Time();
	resDTO->professional_cate = resD0.front().getProfessional_Cate();
	resDTO->jobtitle_grades = resD0.front().getJobtitle_Grades();
	resDTO->title_employment_time = resD0.front().getTitle_Employment_Time();
	resDTO->org_name = resD0.front().getOrg_Name();
	resDTO->issuing_authority = resD0.front().getIssuing_Authority();
	resDTO->judging_unit = resD0.front().getJudging_Unit();
	resDTO->b_highest_professional_title = resD0.front().getB_Highest_Professional_Title();
	return resDTO;
}

bool JobTitleInfoService::updateData(const JobTitleDTO::Wrapper& dto)
{
	JobTitleInfoDAO dao;
	JobTitleDO data;
	data.setId(dto->id);
	data.setEmployee_Id(dto->employee_id);
	data.setEmployee_Name(dto->employee_name);
	data.setEmployee_State(dto->employee_state);
	data.setOrg_Name(dto->org_name);
	data.setJobtitle_Name(dto->jobtitle_name);
	data.setGet_Time(dto->get_time);
	data.setCertificate_Id(dto->certificate_id);
	data.setProfessional_Cate(dto->professional_cate);
	data.setJobtitle_Grades(dto->jobtitle_grades);
	data.setTitle_Employment_Time(dto->title_employment_time);
	data.setIssuing_Authority(dto->issuing_authority);
	data.setJudging_Unit(dto->judging_unit);
	data.setB_Highest_Professional_Title(dto->b_highest_professional_title);
	return !!dao.update(data);
}

File: hr-frontend/src/stores/jobTitleInfo.js
import { defineStore } from 'pinia'

export const useJobTitleInfoStore = defineStore('jobTitleInfo', {
  state: () => ({
    tableTitle: '职称信息',
    tableOperations: [
      {
        name: '搜索'
      },
      {
        name:'新增'
      }
    ],
    xmlData: [
      { id: 1, name: '员工编号', prop: 'employee_id' },
      { id: 2, name: '员工姓名', prop: 'employee_name' },
      { id: 3, name: '员工状态', prop: 'employee_state' },
      { id: 4, name: '组织', prop: 'org_name' },
      { id: 5, name: '职称名称', prop: 'jobtitle_name' },
      { id: 6, name: '获取日期', prop: 'get_time' },
      { id: 7, name: '证书编号', prop: 'certificate_id' },
      { id: 8, name: '专业类别', prop: 'professional_cate' },
      { id: 9, name: '职称等级', prop: 'jobtitle_grades' },
      { id: 10, name: '职称获取月数', prop: 'employment_month' },
      { id: 11, name: '职称聘用时间', prop: 'title_employment_time' },
      { id: 12, name: '签发机构', prop: 'issuing_authority' },
      { id: 13, name: '评审单位', prop: 'judging_unit' },
      { id: 14, name: '是否最高职称', prop: 'b_highest_professional_title' }
    ],
    tableData: null,
    pageSizes: ['10', '20'],
    total: 1000
  }),
  actions: {
    initTableData() {
      this.tableData = [
        {
          id: 1,
          name: '施工员'
        },
        {
          id: 2,
          name: '安全员'
        }
      ]
    },
    addData(val) {
      this.tableData.push(val)
    }
  }
})
